index.js

```diff
@@ -260,24 +279,16 @@ EventEmitter.prototype.removeListener = function removeListener(event, fn, conte
  * Remove all listeners, or those of the specified event.
  *
  * @param {String|Symbol} [event] The event name.
- * @returns {EventEmitter} `this`.
+ * @returns {EventEmitter} Reference to the EventEmitter instance.
  * @api public
  */
 EventEmitter.prototype.removeAllListeners = function removeAllListeners(event) {
-  var evt;
-
-  if (event) {
-    evt = prefix ? prefix + event : event;
-    if (this._events[evt]) {
-      if (--this._eventsCount === 0) this._events = new Events();
-      else delete this._events[evt];
-    }
-  } else {
+  if (!event) {
     this._events = new Events();
     this._eventsCount = 0;
+    return this;
   }
-
-  return this;
+  return this.removeListener(event);
```

Comment:
On second thought I think this isn't a good idea as we are now incurring in all the overhead need to filter the listeners (the for loop).

I bet `ee.removeAllListeners('foo')` is way slower now.